### DISCUSSION TITLE
When refreshing the mounted static files of react app, it is giving 404 error, by calling the frontend url to the backend

### DISCUSSION THREAD

**Discussion by pip-install-skills** [fastapi/fastapi#11502] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.get("/")
async def root():
    return FileResponse("client/dist/index.html")

app.mount("/", StaticFiles(directory="client/dist/"), name="ui")

if __name__ == "__main__":
    import uvicorn
    uvicorn.run("main:app", host='0.0.0.0', port=int(os.getenv("PORT")), workers=int(os.getenv("WORKERS")), reload=False)
```

### Description

I mounted the react dist folder to my fastapi service, when I try to reload the frontend page, it is calling the frontend endpoint to the backend service like this 

"INFO:     127.0.0.1:55563 - "GET /dashboard HTTP/1.1" 404 Not Found"

/dashboard is the endpoint of frontend, I need to follow the flow of the application to not get any issues, but when I try to refresh the page it is giving 404 error for all pages

### Operating System

Windows

### FastAPI Version

0.110.2

### Pydantic Version

2.7.1

### Python Version

Python 3.11.5

**Comment by n8sty** [comment]
Provide a minimally reproducible example and someone might be able to give you some help

**Comment by mauroSoliz98** [comment]
I have the same problem :crying_cat_face:

**Reply by pip-install-skills** [reply, marked as answer]
check this link,

https://github.com/phanitallapudi/Python-Code-Template/blob/main/mount-frontend-on-fastapi-server.py

you need to add line of code to your main.py file

```python
@app.exception_handler(404)
async def exception_404_handler(request, exc):
    return FileResponse("client/dist/index.html")
```
